lib/utils.nix

```diff
@@ -172,6 +172,41 @@ rec {
     in
     lib.literalExpression exp;
 
+  /**
+    Convert one or several conditional warnings to a final warning list.
+
+    # Example
+
+    ```nix
+    warnings = mkWarnings [
+      {
+        condition = plugins.foo.settings.barIntegration && (!plugins.bar.enable);
+        message = "`barIntegration` is enabled but the `bar` plugin is not."
+      }
+    ];
+    ```
+
+    # Type
+    ```
+    mkWarnings :: String -> List -> List
+    ```
+  */
+  mkWarnings =
+    scope: w:
+    let
+      warnings = if !lib.isList w then [ w ] else w;
+
+      processWarning =
+        warning:
+        lib.optional (warning.condition or true) (
```

Comment:
Q: should the condition be _inverted_ so that it matches assertion-style logic?

I always get confused when assertions have negative conditions and warnings have positive conditions.

Although "normal" warnings having positive conditions while nixvim-warnings having negative conditions would _also_ be confusing.

Reply:
One way we could solve this is by having two well-known condition attrs: e.g. `expect` and `unless` or `when` and `whenNot`, etc

Reply:
> One way we could solve this is by having two well-known condition attrs: e.g. `expect` and `unless` or `when` and `whenNot`, etc

I kinda like this... if we're going to create our own helper might as well make it feel more robust. I think we either have it follow upstream if just a plain condition or allow defining both postive/negative cases.

Reply:
Could do something like:
```nix
processCondition = warning: warning.when or !(warning.expect or false);
```
?

Reply:
> Maybe we can start with that and improve it later.

Agreed, if we feel the need for more options we can add them at the time.